setup.py

```diff
@@ -44,18 +44,15 @@
     ],
     extras_require={
         "test": [
-            "black",
             "botocore",
-            "flake8<=5.0.4",
-            "flake8-rst-docstrings",
-            "isort",
             "jsonschema==3.2.0",
             "pre-commit",
-            "pylint",
+            "pydoclint==0.3.8",
```

Comment:
Why do we need this pin?

Reply:
This is pinned in case there are later regressions. As this is a newer tool and we have a stable version which satisfies our needs, I have left this pinned. The dependency tree for the package is minimal and should not prevent other packages from upgrading. Later on, I will explore unpinning this.

Reply:
No this was an idea I floated with @krneta offline. If we are comfortable with unpinning this dependency and seeing the behavior, it can be done later on.